lib/dams/controller_helper.rb

```diff
@@ -903,6 +903,37 @@ def metadata_display?(data)
       data.any? { |t| t.include?('localDisplay') || t.include?('metadataDisplay') }
     end
 
+    def total_count(collection_id)
+      solr_params = build_params("collections_tesim:#{collection_id}")
+      raw_solr(solr_params).response['numFound']
+    end
+
+    def metadata_obj_count(collection_id)
+      solr_params = build_params("collections_tesim:#{collection_id}", metadata_only_fquery)
+      raw_solr(solr_params).response['numFound']
+    end
+
+    def build_params(query, fquery = nil)
+      { q: query, fq: fquery, rows: 1 }
+    end
+
+    def metadata_only_fquery
+      '(otherRights_tesim:localDisplay OR otherRights_tesim:metadataDisplay OR license_tesim:localDisplay)'
+    end
+
+    def raw_solr(params = {})
```

Comment:
It seems like it's the same method as  https://github.com/ucsdlib/damspas/pull/494/files#diff-956e46b33fb5307990c4e1a4b5fd86ccR345, which need cleanup. But I am not sure why we need it since blacklight already provides a Solr helper function [SolrHelper:get_search_results](https://www.rubydoc.info/gems/blacklight/5.8.2/Blacklight%2FSolrHelper:get_search_results) to do it thought.

Reply:
@lsitu - I use the get_search_results before and it always returns different numFound.  https://github.com/ucsdlib/damspas/pull/494/files#diff-c0623f718c6b24f20f5334d0d4eaf8d5L45
I see this existing function like lookup_unit_name also uses raw_solr()
https://github.com/ucsdlib/damspas/blob/develop/app/controllers/catalog_controller.rb#L336
I use it and can see that it returns the correct numFound for my function.

Reply:
Yep, it seems like a long history for that method. Just curios about it. But I think get_search_results should work as expected for the counts: https://github.com/ucsdlib/damspas/blob/develop/app/controllers/dams_units_controller.rb#L27.

Reply:
@VivianChu I did some tests with your commit to use get_search_results and I do see the issue that it failed to retrieve the objects in catalog controller with a simple query to retrieve all the records in a collection like {:q=>"collections_tesim:xx00063001", :rows=>1}.
Maybe we just clean up the duplicate definition in https://github.com/ucsdlib/damspas/pull/494/files#diff-956e46b33fb5307990c4e1a4b5fd86ccR345 and use raw_solr to avoid that annoying issue?

Reply:
@lsitu - Sounds good.  The code has been updated to remove the duplicate definition and use raw_solr. Thanks